core/src/main/java/org/kohsuke/stapler/StaplerResponse.java

```diff
@@ -173,10 +175,30 @@
      * 
      * <p>As of 1.146, the {@code tree} parameter may be used to control the output
      * in detail; see {@link NamedPathPruner#NamedPathPruner(String)} for details.
+     *
+     * @deprecated Use {@link #serveExposedBean(StaplerRequest, Object, ExportConfig)}
      */
+    @Deprecated
     void serveExposedBean(StaplerRequest req, Object exposedBean, Flavor flavor) throws ServletException,IOException;
 
     /**
+     * Serves the exposed bean in the specified flavor.
+     *
+     * <p>
+     * This method performs the complete output from the header to the response body.
+     * If the flavor is JSON, this method also supports JSONP via the {@code jsonp} query parameter.
+     *
+     * <p>The {@code depth} parameter may be used to specify a recursion depth
+     * as in {@link Model#writeTo(Object,int,DataWriter)}
+     *
+     * <p>As of 1.146, the {@code tree} parameter may be used to control the output
+     * in detail; see {@link NamedPathPruner#NamedPathPruner(String)} for details.
+     *
+     * <p> {@link ExportConfig} is passed by the caller to control serialization behavior
+     */
+    void serveExposedBean(StaplerRequest req, Object exposedBean, ExportConfig exportConfig) throws ServletException,IOException;
```

Comment:
FWIW, this change is causing compile errors in some proprietary plugin. Wondering if we couldn't have leveraged a `default method` here to avoid it. 

Like for instance:

```java
serveExposedBean(req, exposedBean, config.getFlavor());
```

WDYT @jglick @vivek ?